File: frontend/src/api/hotspot.ts
/**
 * 热点追踪 API
 */
import request from './request'

// 热点条目
export interface HotspotItem {
  title: string
  url?: string
  hot?: number
  index?: number
  mobile_url?: string
}

// 热点列表响应
export interface HotspotListResponse {
  platform: string
  platform_name: string
  update_time?: string
  items: HotspotItem[]
}

// 分类信息
export interface CategoryInfo {
  code: string
  name: string
  order: number
}

// 分类列表响应
export interface CategoryListResponse {
  categories: CategoryInfo[]
}

// 平台信息
export interface PlatformInfo {
  code: string
  name: string
  category: string // 所属分类
  icon?: string
  color?: string
  subtypes?: Record<string, string> // 子类型（如百度的热搜/汽车/游戏等）
}

// 平台列表响应
export interface PlatformListResponse {
  platforms: PlatformInfo[]
}

// 选题建议请求
export interface TopicSuggestRequest {
  hot_title: string
  url?: string
  user_domain?: string
  target_platforms?: string[]
  model_id?: number
}

// 创作角度
export interface WritingAngle {
  angle: string
  title_suggestion: string
  content_direction: string
  recommended_tools: string[]
  target_audience: string
}

// 选题建议响应
export interface TopicSuggestResponse {
  hot_title: string
  background: string
  angles: WritingAngle[]
  keywords: string[]
  model?: {
    name: string
    provider: string
    model_name: string
  }
  is_fallback?: boolean
}

/**
 * 获取热点分类列表
 */
export function getCategories() {
  return request.get<CategoryListResponse>('/v1/hotspot/categories')
}

/**
 * 获取支持的热点平台列表
 */
export function getPlatforms() {
  return request.get<PlatformListResponse>('/v1/hotspot/platforms')
}

/**
 * 获取指定平台的热点列表
 * @param platform 平台代码
 * @param limit 返回数量
 * @param type 子类型（百度专用：realtime/car/game/movie/novel/teleplay）
 */
export function getHotList(platform: string, limit: number = 20, type?: string) {
  return request.get<HotspotListResponse>('/v1/hotspot/list', {
    params: { platform, limit, type },
  })
}

/**
 * 获取多个平台的热点列表
 * @param platforms 平台代码数组
 * @param limit 每个平台返回数量
 */
export function getMultiPlatformHotList(platforms: string[], limit: number = 10) {
  return request.get<HotspotListResponse[]>('/v1/hotspot/multi', {
    params: {
      platforms: platforms.join(','),
      limit,
    },
  })
}

/**
 * 获取 AI 选题建议
 * @param data 选题建议请求
 */
export function getTopicSuggestions(data: TopicSuggestRequest) {
  return request.post<TopicSuggestResponse>('/v1/hotspot/suggest', data)
}

// 提取关键词响应
export interface ExtractKeywordsResponse {
  title: string
  keywords: string[]
  additional_description?: string
}

/**
 * 从热点标题中提取关键词
 * @param title 热点标题
 * @param url 可选的热点链接
 */
export function extractKeywords(title: string, url?: string) {
  return request.post<ExtractKeywordsResponse>('/v1/hotspot/extract-keywords', null, {
    params: { title, url },
  })
}

File: backend/app/schemas/hotspot.py
"""
热点追踪相关 Schema
"""
from typing import List, Optional, Dict
from pydantic import BaseModel, Field


class HotspotItem(BaseModel):
    """热点条目"""
    title: str = Field(..., description="热点标题")
    url: Optional[str] = Field(None, description="热点链接")
    hot: Optional[int] = Field(None, description="热度值")
    index: Optional[int] = Field(None, description="排名")
    mobile_url: Optional[str] = Field(None, description="移动端链接")


class HotspotListResponse(BaseModel):
    """热点列表响应"""
    platform: str = Field(..., description="平台名称")
    platform_name: str = Field(..., description="平台中文名")
    update_time: Optional[str] = Field(None, description="更新时间")
    items: List[HotspotItem] = Field(default_factory=list, description="热点列表")


class CategoryInfo(BaseModel):
    """分类信息"""
    code: str = Field(..., description="分类代码")
    name: str = Field(..., description="分类中文名")
    order: int = Field(..., description="排序顺序")


class CategoryListResponse(BaseModel):
    """分类列表响应"""
    categories: List[CategoryInfo] = Field(..., description="分类列表")


class PlatformInfo(BaseModel):
    """平台信息"""
    code: str = Field(..., description="平台代码")
    name: str = Field(..., description="平台中文名")
    category: str = Field(..., description="所属分类")
    icon: Optional[str] = Field(None, description="平台图标")
    color: Optional[str] = Field(None, description="平台主题色")
    subtypes: Optional[Dict[str, str]] = Field(None, description="子类型（如百度的热搜/汽车/游戏等）")


class PlatformListResponse(BaseModel):
    """平台列表响应"""
    platforms: List[PlatformInfo] = Field(..., description="支持的平台列表")


class TopicSuggestRequest(BaseModel):
    """选题建议请求"""
    hot_title: str = Field(..., description="热点标题")
    url: Optional[str] = Field(None, description="可选的热点链接，用于获取内容生成更精准的选题建议")
    user_domain: Optional[str] = Field(None, description="用户领域（如：科技、娱乐、职场等）")
    target_platforms: Optional[List[str]] = Field(None, description="目标平台列表")
    model_id: Optional[int] = Field(None, description="指定使用的AI模型ID（不传则用默认模型）")


class SuggestionModelInfo(BaseModel):
    """选题建议使用的模型信息"""
    name: str = Field(..., description="模型名称")
    provider: str = Field(..., description="供应商")
    model_name: str = Field(..., description="模型标识")


class WritingAngle(BaseModel):
    """创作角度"""
    angle: str = Field(..., description="创作角度描述")
    title_suggestion: str = Field(..., description="标题建议")
    content_direction: str = Field(..., description="内容方向")
    recommended_tools: List[str] = Field(..., description="推荐的写作工具类型")
    target_audience: str = Field(..., description="目标受众")


class TopicSuggestResponse(BaseModel):
    """选题建议响应"""
    hot_title: str = Field(..., description="原热点标题")
    background: str = Field(..., description="热点背景分析")
    angles: List[WritingAngle] = Field(..., description="创作角度列表")
    keywords: List[str] = Field(..., description="相关关键词")
    model: Optional[SuggestionModelInfo] = Field(None, description="使用的AI模型信息")
    is_fallback: bool = Field(False, description="是否为模板兜底建议（AI 分析失败/未登录）")


class ExtractKeywordsRequest(BaseModel):
    """提取关键词请求"""
    title: str = Field(..., description="热点标题")


class ExtractKeywordsResponse(BaseModel):
    """提取关键词响应"""
    title: str = Field(..., description="原标题")
    keywords: List[str] = Field(..., description="提取的关键词列表")
    additional_description: str = Field(default="", description="补充说明，包含内容摘要和参考链接")
